src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/versioncontrol/CodeFileExtractorAndMaintainer.java

```diff
@@ -65,11 +68,52 @@ private void extractAllChangedFiles(Diff diff) {
 				}
 				Diff diffForFile = new Diff();
 				diffForFile.addChangedFile(changedFile);
-				for (KnowledgeElement element : gitExtract.getElementsFromCode(diffForFile)) {
+				KnowledgeElement currentIssue = null;
+				KnowledgeElement currentAlternativeOrDecision = null;
+				List<KnowledgeElement> elements = gitExtract.getElementsFromCode(diffForFile);
+
+				for (KnowledgeElement element : elements) {
 					KnowledgeElement elementInGraph = KnowledgeGraph.getOrCreate(projectKey)
-							.addVertexNotBeeingInDatabase(element);
-					Link link = new Link(source, elementInGraph);
-					KnowledgeGraph.getOrCreate(projectKey).addEdgeNotBeeingInDatabase(link);
+							.addVertexNotBeingInDatabase(element);
+					Link link = new Link();
+					if (element.getType() == KnowledgeType.ISSUE) { // An issue is linked directly to the code file.
+						link = new Link(source, elementInGraph);
```

Comment:
In the future, we should move this into a separate method/class (e.g. AutomaticLinkCreator class), but for now it's ok.